modules/runner/src/integrationTest/java/org/apache/ignite/internal/compute/ItLogicalTopologyTest.java

```diff
@@ -337,6 +339,27 @@ public void onNodeValidated(LogicalNode validatedNode) {
         }
     }
 
+    @Test
+    void nodeLeavesLogicalTopologyOnStop() throws Exception {
+        IgniteImpl entryNode = node(0);
+
+        IgniteImpl secondIgnite = startNode(1);
+
+        entryNode.logicalTopologyService().addEventListener(listener);
+
+        stopNode(1);
+
+        assertTrue(waitForCondition(() -> !events.isEmpty(), 10_000), "Did not see any events in time");
```

Comment:
```suggestion
        assertTrue(waitForCondition(() -> !events.isEmpty(), TimeUnit.SECONDS.toMillis(10)), "Did not see any events in time");
```
Do we have to wait 10 seconds?

Reply:
Applied the change.

If the test passes, the wait will be satisfied in much less than 10 seconds. If the logic is broken, the test will take 10 seconds (not a big deal for a failing test), but we'll not see false positives due to a too short timeout.